### PR TITLE
feat: relations no dp

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Adds ability to connect relations from and to non DP enabled content types 

Will highlight some examples:
`Shop (Non DP) -> Product (DP)`

- You can connect products of a specific status:
```ts
// POST /api::shop.shop/
{
	products: [ { documentId: 'product-1' , locale: 'en', status: 'draft' } ]
}
```

- You can connect to multiple status of the same document at the same time
```ts
// POST /api::shop.shop/
{
	products: [ 
		{ documentId: 'product-1' , locale: 'en', status: 'draft' },
		{ documentId: 'product-1' , locale: 'en', status: 'published' } 
	]
}
```

- By default it will connect to both draft and publish status
```ts
// POST /api::shop.shop/
{
	products: [ 
		// Connects to both draft and publish versions if they exist
		{ documentId: 'product-1' , locale: 'en'} 
	]
}
```

### TODO

Tests (DP -> Non DP) (Non DP -> Non DP) content type relations

### CODE REVIEW

**Comment by joshuaellis** [comment]
This makes senses, i'm worried this is not documented anywhere though, because it's quite important. Thoughts?

**Comment by Marc-Roig** [comment]
> This makes senses, i'm worried this is not documented anywhere though, because it's quite important. Thoughts?

@joshuaellis  completely, this + behaviours with i18n is what I plan to document now, I expect to have something informal ready one day before the relations blitz session. And a proper discussion with piwi afterwards

File: api-tests/core/strapi/document-service/relations/no-dp.test.api.ts [hunk 1]
```diff
@@ -92,15 +92,14 @@ const shopModel = {
       target: PRODUCT_UID,
     },
   },
-  options: {
-    draftAndPublish: false,
-  },
+  draftAndPublish: false,
```
[the remark this answers is not in the document]
Comment (Marc-Roig) [reply api-tests/core/strapi/document-service/relations/no-dp.test.api.ts:95]: I am unaware of the reason, my guess is the CTB UI side needed to send this attribute at the root level of the schema, and not the options. But the UI stores the schema like this, and the CTB backend switches it back to be inside options. We do the same for review workflows.

File: packages/core/core/src/services/document-service/repository.ts [hunk 2]
```diff
@@ -289,7 +276,7 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
 
   async function count(params = {} as any) {
     const query = await pipeAsync(
-      DP.defaultToDraft(contentType),
+      DP.defaultStatus(contentType),
```
[the remark this answers is not in the document]
Comment (Marc-Roig) [reply packages/core/core/src/services/document-service/repository.ts:279]: makes sense! changed defaultStatus to not add any status if D&P is disabled.

At some point I was thinking of refactoring the DP and i18n methods, so they don't have to know if DP and i18n are enabled or not.

It makes naming a bit confusing also, atm:

- `setStatusToDraft` doesn't set it if DP is disabled
- `defaultToDraft`: sets to draft regardless of the content type
- `defaultStatus`: doesn't set it if DP is disabled

I was thinking of keeping `defaultToDraft` and `setStatusToDraft` and delegating the responsability to ignore this action on non DP by doing:
```ts
pipeAsync(
   hasDP ? DP.defaultToDraft : identity
   ...
)
```

wdyt?

File: packages/core/core/src/services/document-service/transform/id-transform.ts [hunk 3]
```diff
@@ -8,35 +11,39 @@ import { transformPopulate } from './populate';
...
+  opts: { [key: string]: any } = {}
+): Promise<Documents.Params.All> {
   // Transform relational documentIds to entity ids
-  let data = input.data;
-  if (input.data) {
-    data = await transformData(input.data, { ...opts, uid });
+  let data = query.data;
+  if (query.data) {
+    data = await transformData(query.data, {
+      ...opts,
```
[the remark this answers is not in the document]
Comment (Marc-Roig) [reply packages/core/core/src/services/document-service/transform/id-transform.ts:21]: nope!

File: packages/core/utils/src/convert-query-params.ts [hunk 4]
```diff
@@ -604,13 +610,32 @@ const convertAndSanitizeFilters = (filters: FiltersParams, schema?: Model): Wher
   return filters;
 };
 
+const convertStatusParams = (status?: 'draft' | 'published', query: Query = {}) => {
```
[the remark this answers is not in the document]
Comment (Marc-Roig) [reply packages/core/utils/src/convert-query-params.ts:613]: will do that in another pr! so we can have the disable dp ready by tomorrowww

File: packages/core/database/src/entity-manager/regular-relations.ts [hunk 5]
```diff
@@ -23,6 +23,36 @@ declare module 'knex' {
   }
 }
 
+//  TODO: This is a short term solution, to not steal relations from the same document.
+const getDocumentSiblingIdsQuery = (con: Knex, tableName: string, id: ID) => {
+  // Find if the model is a content type or something else (e.g component)
+  // to only get the documentId if it's a content type
+  const models: Model[] = Array.from(strapi.db.metadata.values());
```
Comment (alexandrebodin) [inline packages/core/database/src/entity-manager/regular-relations.ts:30]: probably best to add the model in the deleteRelation & othe relational methods signature eventually (let's refacto later)

**Review by alexandrebodin — APPROVED** [review]
Good to merge in the main branch to make the experimental :) I'll do a global re-read of the entire PR after

**Pull request merged**